python/cudf/cudf/core/frame.py

```diff
@@ -561,6 +561,31 @@ def where(self, boolean_mask, other):
             result.index = self.index
             return result
 
+    def _partition(self, scatter_map, npartitions, keep_index=True):
+
+        output_table, output_offsets = libcudfxx.partitioning.partition(
+            self, scatter_map, npartitions, keep_index
+        )
+
+        # due to the split limitation mentioned
+        # here: https://github.com/rapidsai/cudf/issues/4607
+        # we need to remove first & last elements in offsets.
+        # TODO: Remove this after the above issue is fixed.
+        output_offsets = output_offsets[1:-1]
+
+        result = libcudfxx.copying.table_split(
+            output_table, output_offsets, keep_index=keep_index
+        )
+
+        result = [self.__class__._from_table(tbl) for tbl in result]
+
+        [
+            frame._copy_categories(self, include_index=keep_index)
+            for frame in result
+        ]
```

Comment:
@kkraus14 gets on me for writing things as comprehensions like this, so suggesting the alternative :-)

```suggestion
        for frame in result:
            frame._copy_categories(self, include_index=keep_index)
```